### ISSUE TITLE
Inconsistent errors for invalid subcommands and flags

### ISSUE THREAD

**Issue by mohammedzee1000** [spf13/cobra#1409]
We can observe inconsistent behavior with command error when incorrect subcommands are used with or without flags.
This has been observed on https://github.com/openshift/odo/issues/4674

**Comment by johnSchnake** [comment]
That issue also links to #926 which shows some inconsistency for the same command when the args/flags are moved around.

The issue (so people dont have to follow the links) is that `root <invalid command> <flag>` is reporting an error due to the unrecognized flag (instead of the invalid command). Presumably this this has to do with a few things:
 - order of flag parsing/validation
 - mistakingly thinking that an invalid command may be an argument for the parent command instead

**Comment by ardaguclu** [comment]
parentcmd subinvalidcmd --invalid-arg

When subcommand is invalid(`subinvalidcmd`), [findNext](https://github.com/spf13/cobra/blob/bf6cb5804d7a45faacd0abfc99145c42696ec818/command.go#L644) returns nil. Thus, it proceeds with the parent command(`parentcmd`) and fails in [here](https://github.com/spf13/cobra/blob/bf6cb5804d7a45faacd0abfc99145c42696ec818/command.go#L795).

Maybe `findNext` [function](https://github.com/spf13/cobra/blob/bf6cb5804d7a45faacd0abfc99145c42696ec818/command.go#L675) should return error when subcommand not found and we immediately throw "unknown command"?

**Comment by mohammedzee1000** [comment]
> For example;
> 
> ```
> parentcmd subinvalidcmd --invalid-arg
> ````
> 
> When subcommand is invalid(`subinvalidcmd`), [findNext](https://github.com/spf13/cobra/blob/bf6cb5804d7a45faacd0abfc99145c42696ec818/command.go#L644) returns nil. Thus, it proceeds with the parent command(`parentcmd`) and fails in [here](https://github.com/spf13/cobra/blob/bf6cb5804d7a45faacd0abfc99145c42696ec818/command.go#L795).
> 
> Maybe `findNext` [function](https://github.com/spf13/cobra/blob/bf6cb5804d7a45faacd0abfc99145c42696ec818/command.go#L675) should return error when subcommand not found and we immediately throw "unknown command"?

Agreed especially if a command has subcommands defined. I cannot think of situation where someone would want both subcommands and args
